hack/generator/pkg/codegen/pipeline_name_types_for_crd.go

```diff
@@ -93,11 +94,11 @@ func nameInnerTypes(
 	visitor.VisitResourceType = func(this *astmodel.TypeVisitor, it *astmodel.ResourceType, ctx interface{}) astmodel.Type {
 		nameHint := ctx.(string)
 
-		spec := this.Visit(it.SpecType(), nameHint+"Spec")
+		spec := this.Visit(it.SpecType(), nameHint+"_Spec")
```

Comment:
The names of the types only show up in the type graph which isn't exposed in the CRDs so this isn't that big of a deal, but: why `_` + `X`? Isn't the standard idiom in Go to avoid `_` in type names?

Reply:
This is something to be resolved later (#162), at the moment just using `Status` as a suffix without underscore causes some typenames to clash so this was a quick fix for that.